doc/ref/states/requisites.rst

```diff
@@ -482,6 +608,37 @@ if any of the watched states changes.
             - onchanges:
               - file: /etc/myservice/myservice.conf
 
+.. _requisites-onchanges_any:
+
+onchanges_any
+~~~~~~~~~~~~~
+
+.. versionadded:: Oxygen
+
+The ``onchanges_any`` requisite makes a state only apply one of the required states
+generates changes, and if one of the watched state's "result" is ``True``. This can be
+a useful way to execute a post hook after changing aspects of a system.
+
+.. code-block:: yaml
+
+    myservice:
+      pkg.installed:
+        - name: myservice
+        - name: yourservice
+      file.managed:
+        - name: /etc/myservice/myservice.conf
+        - source: salt://myservice/files/myservice.conf
+        - mode: 600
+      file.managed:
+        - name: /etc/yourservice/yourservice.conf
+        - source: salt://yourservice/files/yourservice.conf
+        - mode: 600
+      cmd.run:
+        - name: /usr/libexec/myservice/post-changes-hook.sh
+        - onchanges:
+          - file: /etc/myservice/myservice.conf
+          - file: /etc/your_service/yourservice.conf
+
```

Comment:
Same comment here about adding an example of what the expected state run will be.